src/main/java/com/KacFlor/ShopSpring/controller/CardDataController.java

```diff
@@ -26,10 +26,31 @@ public CardDataController(CardDataService cardDataService){
         this.cardDataService = cardDataService;
     }
 
-    @PreAuthorize("hasAuthority('" + Role.Fields.USER + "')")
+    @PreAuthorize("hasAuthority('" + Role.Fields.ADMIN + "')")
     @GetMapping
-    public List<CardData> getCardData(){
-        return this.cardDataService.getCardData();
+    public ResponseEntity<List<CardData>> getAll(){
+        List<CardData> cardData = cardDataService.getAll();
+        return ResponseEntity.ok(cardData);
     }
 
+    @PreAuthorize("hasAnyAuthority('" + Role.Fields.USER + "', '" + Role.Fields.ADMIN + "')")
+    @GetMapping("/{id}")
+    public ResponseEntity<Optional<CardData>> getById(@PathVariable("id") Integer id){
+        Optional<CardData> cardData = Optional.ofNullable(cardDataService.getCardDataById(id));
+        return ResponseEntity.ok(cardData);
+    }
+
+    @PreAuthorize("hasAnyAuthority('" + Role.Fields.USER + "', '" + Role.Fields.ADMIN + "')")
+    @PatchMapping("/{CDid}/customer/{Cid}")
```

Comment:
why no {id}?

Reply:
because i have 2 id one for CardData (CDid) and one for Customer (Cid)